src/script/components/Conversation/ConversationCells/CellsPagination/CellsPagination.tsx

```diff
@@ -0,0 +1,175 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React, {useCallback} from 'react';
+
+import {FlexBox, Bold, Link, IconButton, Select} from '@wireapp/react-ui-kit';
+
+import * as Icon from 'Components/Icon';
+
+import {
+  pagesContainerStyles,
+  numberStyles,
+  numberActiveStyles,
+  arrowButtonStyles,
+  arrowNextIconStyles,
+  arrowPreviousIconStyles,
+  containerStyles,
+  selectorStyles,
+  pageSelectorContainer,
+} from './CellsPagination.styles';
+
+import {useCellsStore} from '../common/useCellsStore/useCellsStore';
+
+interface CellsPaginationProps {
+  currentPage?: number;
+  numberOfPages?: number;
+  firstRow?: number;
+  lastRow?: number;
+  totalRows?: number;
+  goPage: (pageIndex: number) => void;
+  [key: string]: any;
+}
+
+export const CellsPagination: React.FC<CellsPaginationProps> = ({
+  currentPage = 0,
+  numberOfPages = 1,
+  goPage,
+  firstRow = 1,
+  lastRow,
+  totalRows,
+  ...props
+}) => {
+  const isLastPage = currentPage === numberOfPages - 1;
+  const isFirstPage = currentPage === 0;
+
+  const {pageSize, setPageSize} = useCellsStore();
+
+  const onSizeChange = useCallback(
+    (value: {value: string; label: string}) => {
+      setPageSize(parseInt(value.value));
+    },
+    [setPageSize],
+  );
+
+  const renderPageList = () => {
+    const lastPageIndex = numberOfPages - 1;
+    const spanLength = 1;
+    const endLength = 1;
+    const skipLength = 1;
+    const normalizeCount = endLength + skipLength + spanLength;
+
+    const dots = (key: string) => (
+      <Bold css={numberStyles} key={key}>
+        …
+      </Bold>
+    );
+
+    const renderPageNumber = (pageIndex: number) =>
+      currentPage === pageIndex ? (
+        <Bold key={pageIndex} css={{...numberStyles, ...numberActiveStyles}} data-uie-name="status-active-page">
+          {pageIndex + 1}
+        </Bold>
+      ) : (
+        <Link css={numberStyles} key={pageIndex} onClick={() => goPage(pageIndex)} data-uie-name="go-page">
+          {pageIndex + 1}
+        </Link>
+      );
+
+    const normalizedCurrent = Math.min(Math.max(currentPage, normalizeCount), lastPageIndex - normalizeCount);
+    const beforeCount = normalizedCurrent - spanLength - endLength;
+    const afterCount = lastPageIndex - endLength - normalizedCurrent - spanLength;
+
+    const pages = Array.from({length: numberOfPages}, (_, index) => renderPageNumber(index));
+
+    if (afterCount > skipLength) {
+      pages.splice(normalizedCurrent + spanLength + 1, afterCount, dots('dots-end'));
+    }
+    if (beforeCount > skipLength) {
+      pages.splice(endLength, beforeCount, dots('dots-start'));
+    }
+
+    return pages;
+  };
+
+  const options = [
+    {value: '10', label: '10'},
+    {value: '20', label: '20'},
+    {value: '50', label: '50'},
+    {value: '100', label: '100'},
+  ];
+  const crtValue = {value: `${pageSize}`, label: `${pageSize}`};
```

Comment:
what `crt` mean? maybe we can name it differently and more self-descriptive?

Reply:
renamed to currentOption